src/mono/browser/runtime/http.ts

```diff
@@ -66,167 +81,180 @@ export function http_wasm_abort_response(res: ResponseExtension): void {
+    if (BuildConfiguration === "Debug") commonAsserts(controller);
     // the bufferPtr is pinned by the caller
     const view = new Span(bufferPtr, bufferLength, MemoryViewType.Byte);
     return wrap_as_cancelable_promise(async () => {
-        if (!res.__reader) {
-            res.__reader = res.body!.getReader();
+        if (!controller.streamReader) {
+            controller.streamReader = controller.fetchResponse!.body!.getReader();
         }
-        if (!res.__chunk) {
-            res.__chunk = await res.__reader.read();
-            res.__source_offset = 0;
+        if (!controller.currentStreamReaderChunk) {
+            controller.currentStreamReaderChunk = await controller.streamReader.read();
+            controller.currentBufferOffset = 0;
         }
-        if (res.__chunk.done) {
+        if (controller.currentStreamReaderChunk!.done) {
             return 0;
         }
 
-        const remaining_source = res.__chunk.value.byteLength - res.__source_offset;
+        const remaining_source = controller.currentStreamReaderChunk.value.byteLength - controller.currentBufferOffset;
         mono_assert(remaining_source > 0, "expected remaining_source to be greater than 0");
 
         const bytes_copied = Math.min(remaining_source, view.byteLength);
-        const source_view = res.__chunk.value.subarray(res.__source_offset, res.__source_offset + bytes_copied);
+        const source_view = controller.currentStreamReaderChunk.value.subarray(controller.currentBufferOffset, controller.currentBufferOffset + bytes_copied);
         view.set(source_view, 0);
-        res.__source_offset += bytes_copied;
+        controller.currentBufferOffset += bytes_copied;
         if (remaining_source == bytes_copied) {
-            res.__chunk = undefined;
+            controller.currentStreamReaderChunk = undefined;
         }
 
         return bytes_copied;
     });
 }
 
-interface TransformStreamExtension extends TransformStream<Uint8Array, Uint8Array> {
-    __writer: WritableStreamDefaultWriter<Uint8Array>
-    __fetch_promise?: Promise<ResponseExtension>
-}
+interface HttpController {
+    abortController: AbortController
+
+    // streaming request
+    streamReader?: ReadableStreamDefaultReader<Uint8Array>
+
+    // response
+    fetchResponsePromise?: ControllablePromise<Response>
+    fetchResponse?: Response
+    responseHeaderNames: string[];
+    responseHeaderValues: string[];
+    currentBufferOffset: number
+
+    // non-streaming response
+    responseBuffer?: ArrayBuffer
 
-interface ResponseExtension extends Response {
-    __buffer?: ArrayBuffer
-    __reader?: ReadableStreamDefaultReader<Uint8Array>
-    __chunk?: ReadableStreamReadResult<Uint8Array>
-    __source_offset: number
-    __abort_controller: AbortController
-    __headerNames: string[];
-    __headerValues: string[];
+    // streaming response
+    streamWriter: WritableStreamDefaultWriter<Uint8Array>
```

Comment:
Nullable properties are incosistent. To me it looks like everything except `abortController` could/should be nullable.